catalog/rocks_cataloger.go

```diff
@@ -28,8 +28,12 @@ const (
 
 var ErrUnknownDiffType = errors.New("unknown graveler difference type")
 
-func NewCataloger(db db.Database, cfg *config.Config) (Cataloger, error) {
-	entryCatalog, err := NewEntryCatalog(cfg, db)
+func NewCatalogerSingleWriter(db db.Database, cfg *config.Config) (Cataloger, error) {
```

Comment:
Why do we need to special case this? why not always pass 2 pools?

Reply:
Discuss f2f, switched to config struct.